controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/kvtable/CreateTableTask.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.eventProcessor.requesthandlers.kvtable;
+
+import com.google.common.base.Preconditions;
+import io.pravega.controller.store.kvtable.CreateKVTableResponse;
+import io.pravega.controller.store.kvtable.KVTableState;
+import io.pravega.controller.store.kvtable.KVTableMetadataStore;
+import io.pravega.client.tables.KeyValueTableConfiguration;
+import io.pravega.controller.store.kvtable.KVTOperationContext;
+import io.pravega.controller.store.stream.State;
+import io.pravega.controller.task.KeyValueTable.TableMetadataTasks;
+import io.pravega.shared.NameUtils;
+import io.pravega.shared.controller.event.kvtable.CreateTableEvent;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import static io.pravega.controller.task.Stream.TaskStepsRetryHelper.withRetries;
+/**
+ * Request handler for performing scale operations received from requeststream.
+ */
+@Slf4j
+public class CreateTableTask implements TableTask<CreateTableEvent> {
+
+    private final KVTableMetadataStore kvtMetadataStore;
+    private final TableMetadataTasks kvtMetadataTasks;
+    private final ScheduledExecutorService executor;
+
+    public CreateTableTask(final KVTableMetadataStore kvtMetaStore,
+                           final TableMetadataTasks kvtMetaTasks,
+                           final ScheduledExecutorService executor) {
+        Preconditions.checkNotNull(kvtMetaStore);
+        Preconditions.checkNotNull(kvtMetaTasks);
+        Preconditions.checkNotNull(executor);
+        this.kvtMetadataStore = kvtMetaStore;
+        this.kvtMetadataTasks = kvtMetaTasks;
+        this.executor = executor;
+    }
+
+    @Override
+    public CompletableFuture<Void> execute(final CreateTableEvent request) {
+        String scope = request.getScopeName();
+        String kvt = request.getKvtName();
+        int partitionCount = request.getPartitionCount();
+        long creationTime = request.getTimestamp();
+        long requestId = request.getRequestId();
+        String kvTableId = request.getTableId().toString();
+        KeyValueTableConfiguration config = new KeyValueTableConfiguration(partitionCount);
+
+        return kvtMetadataStore.getKVTable(scope, kvt, null).getId()
```

Comment:
even this is a network call so wrapping this in retries block for retryable errors is required.

Reply:
fixed